src/AasxServerStandardBib/Services-DB/AdminShellPackageEnvironmentServiceDB.cs

```diff
@@ -0,0 +1,368 @@
+            Program.signalNewData(2);
+            return body;
+        }
+        public void DeleteAssetAdministrationShell(int packageIndex, IAssetAdministrationShell aas)
+        {
+            bool deleted = _database.DeleteDBAssetAdministrationShellById(aas);
+            if (deleted)
+            {
+                _logger.LogDebug($"Deleted Asset Administration Shell with id {aas.Id}");
+                Program.signalNewData(2);
+            }
+            else
+            {
+                _logger.LogError($"Could not delete Asset Administration Shell with id {aas.Id}");
+            }
+        }
+        public List<IAssetAdministrationShell> GetAllAssetAdministrationShells()
+        {
+            return _database.GetLINQAssetAdministrationShell().ToList().Cast<IAssetAdministrationShell>().ToList();
+        }
+        public IAssetAdministrationShell GetAssetAdministrationShellById(string aasIdentifier, out int packageIndex)
+        {
+            packageIndex = -1; //TODO unused -> remove in the future
+            var output = _database.GetLINQAssetAdministrationShell()
+                .Where(r => r.Id.Equals(aasIdentifier))
+                .ToList();
+
+            if (output.Any())
+            {
+                _logger.LogDebug($"Asset Administration Shell with id {aasIdentifier} found.");
+                return output.First();
+            }
+            else
+            {
+                throw new NotFoundException($"Asset Administration Shell with id {aasIdentifier} not found.");
+            }
+        }
+        public bool IsAssetAdministrationShellPresent(string aasIdentifier)
+        {
+            return _database.GetLINQAssetAdministrationShell()
+                .Where(r => r.Id.Equals(aasIdentifier))
+                .ToList().Any();
+        }
+        public void UpdateAssetAdministrationShellById(IAssetAdministrationShell body, string aasIdentifier)
+        {
+            //TODO Test -> should work -> function unused
+            var timeStamp = DateTime.UtcNow;
+            body.TimeStampCreate = timeStamp;
+            body.SetTimeStamp(timeStamp); 
+            
+            _database.UpdateDBAssetAdministrationShellById(body, aasIdentifier);
+
+            Program.signalNewData(1); //0 not working, hence 1 = same tree, structure may change
+            _logger.LogDebug($"Successfully updated the AAS with requested AAS");
+        }
+        public void ReplaceAssetAdministrationShellById(string aasIdentifier, IAssetAdministrationShell newAas)
+        {
+            var timeStamp = DateTime.UtcNow;
+            newAas.TimeStampCreate = timeStamp;
+            newAas.SetTimeStamp(timeStamp);
+            _database.UpdateDBAssetAdministrationShellById(newAas, aasIdentifier);
+            Program.signalNewData(1);
+        }
+
+        public void DeleteAssetInformationThumbnail(int packageIndex, IResource defaultThumbnail)
+        {
+            //TODO
```

Comment:
Regarding your TODO comments, you have two options:

1. Please resolve the TODO before finalizing this PR.
2. If you cannot resolve it immediately, please add detailed information on why and create a GitHub issue to keep track of it.